tests/test_ddi.py

```diff
@@ -51,3 +62,141 @@ def test_get_variable_info(cps_ddi: ddi.Codebook, cps_df: pd.DataFrame):
     # And does it raise a ValueError if the variable does not exist?
     with pytest.raises(ValueError):
         cps_ddi.get_variable_info("foo")
+
+
+def test_get_all_types(cps_ddi: ddi.Codebook, cps_df: pd.DataFrame):
+
+    var_types = {
+        "YEAR": "numeric",
+        "SERIAL": "numeric",
+        "HWTSUPP": "numeric",
+        "STATEFIP": "numeric",
+        "MONTH": "numeric",
+        "PERNUM": "numeric",
+        "WTSUPP": "numeric",
+        "INCTOT": "numeric",
+    }
+
+    assert cps_ddi.get_all_types(type_format="vartype") == var_types
+
+    python_types = {
+        "YEAR": int,
+        "SERIAL": int,
+        "HWTSUPP": float,
+        "STATEFIP": int,
+        "MONTH": int,
+        "PERNUM": int,
+        "WTSUPP": float,
+        "INCTOT": int,
+    }
+
+    assert cps_ddi.get_all_types(type_format="python_type") == python_types
+    # always np.float64 due to eventual NaNs.
+    numpy_types = {
+        "YEAR": np.float64,
+        "SERIAL": np.float64,
+        "HWTSUPP": np.float64,
+        "STATEFIP": np.float64,
+        "MONTH": np.float64,
+        "PERNUM": np.float64,
+        "WTSUPP": np.float64,
+        "INCTOT": np.float64,
+    }
+
+    assert cps_ddi.get_all_types(type_format="numpy_type") == numpy_types
+
+    pandas_types = {
+        "YEAR": pd.Int64Dtype(),
+        "SERIAL": pd.Int64Dtype(),
+        "HWTSUPP": np.float64,
+        "STATEFIP": pd.Int64Dtype(),
+        "MONTH": pd.Int64Dtype(),
+        "PERNUM": pd.Int64Dtype(),
+        "WTSUPP": np.float64,
+        "INCTOT": pd.Int64Dtype(),
+    }
+
+    assert cps_ddi.get_all_types(type_format="pandas_type") == pandas_types
+
+    pandas_types_efficient = {
+        "YEAR": np.float64,
+        "SERIAL": np.float64,
+        "HWTSUPP": np.float64,
+        "STATEFIP": np.float64,
+        "MONTH": np.float64,
+        "PERNUM": np.float64,
+        "WTSUPP": np.float64,
+        "INCTOT": np.float64,
+    }
+
+    assert (
+        cps_ddi.get_all_types(type_format="pandas_type_efficient")
+        == pandas_types_efficient
+    )
+
+    acceptable_values = [
+        "numpy_type",
+        "pandas_type",
+        "pandas_type_efficient",
+        "python_type",
+        "vartype",
+    ]
+
+    # Does it raise a ValueError if the specified type of format, doesn't match existing attribute?
+    with pytest.raises(ValueError):
+        cps_ddi.get_all_types(type_format="foo")
+
+
+@pytest.mark.character_data
+def test_get_all_types_with_pyarrow(cps_ddi2: ddi.Codebook, cps_df2: pd.DataFrame):
+    # this test can be run with variable INDNAICS from IPUMS USA sample ACS 2020 5% for example. I didn't find
```

Comment:
I have added a small test CPS dataset to the test/fixtures directory that contains a string variable. The string variable is HRSERSUF. Please update these tests to use that file and remove the `/tmp/` file work around from the tests and the gitignore file.